### DISCUSSION TITLE
In Next js 14 App Router,  I can't get path name and as path in app router like pages router. Any suggesetions or methods to achieve that feature

### DISCUSSION THREAD

**Discussion by saijoro** [vercel/next.js#63705] [Help]
**For Example**
next js pages router - 
(router.asPath: "/farm/66016e78bc87d1994aebe41b") and 
(router.pathname: "/farm/[farm_id]") is this 

but in app router using pathname only returns
this one /farm/66016e78bc87d1994aebe41b

then is there any option to get '/farm/[farm_id]' like this, i have much dependency with it. Useful solution can help me 

1. Can we give an alias name to every route like done in vue js, 
such that i can check whether the route exist or not with the alias name it self, 

The main requirement is, i wanna change the heading of the page based in the navbar based on the pathname, which cannot work for dynamic routes, so please help me with this basic problem, 
Note: Mark that the Navbar is common for every page. 

2. And also i want to show navbar in some pages and hide in other some pages, how can we do that in nextjs 14 app router, 
it is the common case for every application. 

here is my layout.ts file - 
```
"use client";
import { Providers } from "@/Redux/Provider";
import SubLayout from "@/components/Auth/Profile/SubLayout/SubLayout";
import Head from "next/head";
import { usePathname } from "next/navigation";
import "../scss/app.scss";
import "./globals.css";

import NavBarComponent from "@/components/NavBar/NavBarComponent";
import {
  isRootLayoutVisible,
  isSubLayoutForProfileVisible,
} from "@/lib/helpers/isLayout";

export default function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  const pathname = usePathname();

  return (
    <html lang="en">
      <Head>
        <title>labsquire</title>
      </Head>
      <body>
        <Providers>
          {isRootLayoutVisible(pathname) ? (
            <NavBarComponent>
              {isSubLayoutForProfileVisible(pathname) ? (
                <SubLayout>{children}</SubLayout>
              ) : (
                children
              )}
            </NavBarComponent>
          ) : (
            children
          )}
        </Providers>
      </body>
    </html>
  );
}
```

and the isLayout function.

```
const noRootLayoutUrls = [
  "/signin-verify",
  "/forgot-password",
  "/result-files/supplementary-report/",
  "/result-files/regular-report/",
  "/case-e-requisition",
  "/multiple-qrcodes",
  "/orders/multiple-barcodes",
  "/multiple-barcodes",
  "/order-e-requisition/",
];
const subLayoutForProfileUrls = ["/profile", "/update-password"];

export const isRootLayoutVisible = (pathname: string) => {
  if (pathname == "/") return false;
  let isVisible;

  for (let i = 0; i <= noRootLayoutUrls?.length; i++) {
    if (pathname.includes(noRootLayoutUrls[i])) {
      isVisible = false;
      break;
    } else {
      isVisible = true;
    }
  }

  return isVisible;
};

export const isSubLayoutForProfileVisible = (pathname: string) => {
  if (subLayoutForProfileUrls.includes(pathname)) {
    return true;
  }
  return false;
};
```

This was i am following which looks awkward, Please help to improve my code standards, it can help most of the beginners like me.

**Comment by Lrunlin** [comment]
这样就行 https://github.com/vercel/next.js/discussions/61846  可以参考一下

function useGetRawPath() {
  const params = useParams();
  const pathname = usePathname();
  const splittedRoute = pathname.split("/");

  if (typeof params === "object" && params !== null) {
    let reconstructedPath = "";

    for (let i = 1; i < splittedRoute.length; i++) {
      const segment = splittedRoute[i];

      // Check if the segment matches any value in the params object
      const matchingKey = Object.keys(params).find(key => params[key] === segment);

      if (matchingKey) {
        // Replace the segment with the dynamic segment
        reconstructedPath += `/[${matchingKey}]`;
      } else {
        // Keep the original segment
        reconstructedPath += `/${segment}`;
      }
    }

    return reconstructedPath;
  } else {
    // Params object is not present
    return pathname;
  }
}
